File: library-java/src/main/java/com/github/shiraji/butai/ButaiDelegate.java
package com.github.shiraji.butai;

import android.app.Application;


public class ButaiDelegate {
    private static ButaiActivityLifecycleCallbacks callback = new ButaiActivityLifecycleCallbacks();

    public static void init(Application application) {
        application.registerActivityLifecycleCallbacks(callback);
    }

    public static boolean isReturnedFromBackground() {
        return callback.isReturnedFromBackground();
    }

    public static boolean isBackground() {
        return callback.isBackground();
    }

    public static boolean isForeground() {
        return callback.isForeground();
    }
}

File: library/src/main/java/com/github/shiraji/butai/ButaiDelegate.kt
package com.github.shiraji.butai

import android.app.Application

object ButaiDelegate : Butai {

    private val callback = ButaiActivityLifecycleCallbacks()

    fun init(application: Application) {
        application.registerActivityLifecycleCallbacks(callback)
    }

    override fun isReturnedFromBackground() = callback.isReturnedFromBackground
    override fun isBackground() = callback.isBackground
    override fun isForeground() = callback.isForeground
}